packages/flutter/test/material/dropdown_menu_test.dart

```diff
@@ -2432,6 +2432,59 @@ void main() {
     final TextField textField = tester.widget(find.byType(TextField));
     expect(textField.keyboardType, TextInputType.text);
   });
+
+  // Regression test for https://github.com/flutter/flutter/pull/151684.
+  testWidgets('DropdownMenu.expandedInsets accepts EdgeInsetsGeometry', (WidgetTester tester) async {
+    const double parentWidth = 500.0;
+    final List<DropdownMenuEntry<ShortMenu>> shortMenuItems = <DropdownMenuEntry<ShortMenu>>[];
+    for (final ShortMenu value in ShortMenu.values) {
+      final DropdownMenuEntry<ShortMenu> entry = DropdownMenuEntry<ShortMenu>(value: value, label: value.label);
+      shortMenuItems.add(entry);
+    }
+    Widget buildMenuAnchor() {
+      return MaterialApp(
+        home: Scaffold(
+          body: SizedBox(
+            width: parentWidth,
+            height: parentWidth,
+            child: DropdownMenu<ShortMenu>(
+              // The DropdownMenu should accept any EdgeInsetsGeometry, not just EdgeInsets.
+              expandedInsets: const EdgeInsetsDirectional.only(start: 35.0, top: 50.0, end: 20.0),
+              dropdownMenuEntries: shortMenuItems,
+            ),
+          ),
+        ),
+      );
+    }
+
+    // By default, the width of the text field is determined by the menu children.
+    await tester.pumpWidget(buildMenuAnchor());
+    RenderBox box = tester.firstRenderObject(find.byType(TextField));
+    expect(box.size.width, 136.0);
+
+    await tester.tap(find.byType(TextField));
+    await tester.pumpAndSettle();
+
+    Size buttonSize = tester.getSize(find.widgetWithText(MenuItemButton, 'I0').hitTestable());
+    expect(buttonSize.width, 136.0);
+
+    // The width of the text field should be adjusted based on the EdgeInsetsDirectional.start and
+    // EdgeInsetsDirectional.right. The top and bottom values will be ignored.
+    await tester.pumpWidget(Container());
+    await tester.pumpWidget(buildMenuAnchor());
```

Comment:
I think there is a bug in the test here? You call `await tester.pumpWidget(buildMenuAnchor());` twice, but the second call will reuse the same EdgeInsetsDirectional value.

You probably meant to pass in `const EdgeInsetsDirectional.only(start: 35.0, top: 50.0, end: 20.0)` as a parameter to `buildMenuAnchor()`, so that the second pump would use a different padding value?

Reply:
Apologies, got my local environment set up finally so confirmed the tests pass.

Ended up just adding a case to an existing test that this was copied from, as it fits in there.